website/docs/r/cloudfront_distribution.html.markdown

```diff
@@ -142,6 +142,100 @@ resource "aws_cloudfront_distribution" "s3_distribution" {
 }
 ```
 
+### CloudFront distribution with custom origin
+
+The example below uses an S3 static web hosting origin for use-cases beyond asset downloads. AWS doesn't allow using the S3 website endpoint from CloudFront. This example works using a custom authentication and a custom origin to further provide caching and HTTPS.
+
+```hcl
+locals {
+  origin_id = "S3-Website-${aws_s3_bucket.website-bucket.bucket}"
+  cloudfront-authentication-user-agent = "somethingRandomEen8Ohta"
+}
+
+resource "aws_s3_bucket" "website-bucket" {
+  bucket = "webhosting-bucket-via-cloudfront"
+
+  website {
+    index_document = "index.html"
+  }
+}
+
```

Comment:
I had been using the actual Terraform data version for my policy, and using bucket ID was rejected by AWS as "invalid resource." I used the ARN and it works fine.

```suggestion
data "aws_iam_policy_document" "bucket-policy" {
  statement {
    actions   = ["s3:GetObject"]
    resources = ["${aws_s3_bucket.website-bucket.arn}/*"]

    principals {
      type        = "*"
      identifiers = ["*"]
    }

    condition {
      test     = "StringEquals"
      variable = "aws:UserAgent"
      values   = ["${local.cloudfront-authentication-user-agent}"]
    }
  }
}
```